### PR TITLE
Include using the event end date when returning events from a range

### PR DESCRIPTION
**Pull request by grimmlink**
Retrieve element having a start or end date outside of range but that should still appear

### CODE REVIEW

File: src/ICal/ICal.php [hunk 1]
```diff
@@ -1270,8 +1270,13 @@ public function eventsFromRange($rangeStart = false, $rangeEnd = false)
...
         foreach ($events as $anEvent) {
-            $timestamp = $anEvent->dtstart_array[2];
-            if ($timestamp >= $rangeStart && $timestamp <= $rangeEnd) {
+            $eventStart = $anEvent->dtstart_array[2];
+            $eventEnd = $anEvent->dtend_array[2];
+            
+            if ($eventStart >= $rangeStart && $eventStart < $rangeEnd)  // event start date contained in the range
+                || ($eventEnd > $rangeStart && $eventEnd <= $rangeEnd)  // event end date contained in the range
+                || ($eventStart < $rangeStart && $eventEnd > $rangeEnd) // event starts before and finishes after range
+            {
```
Comment (u01jmg3) [inline src/ICal/ICal.php]: Please change to `if () {` (PSR-2)

File: src/ICal/ICal.php [hunk 2]
```diff
@@ -1304,4 +1309,4 @@ public function sortEventsWithOrder($events, $sortOrder = SORT_ASC)
 
         return $extendedEvents;
     }
-}
\ No newline at end of file
+}
```
Comment (u01jmg3) [inline src/ICal/ICal.php]: No blank newline

**Comment by u01jmg3** [comment]
Thanks for this

- Can you update the [docblock for `eventsFromRange`](https://github.com/u01jmg3/ics-parser/blob/master/src/ICal/ICal.php#L1206-L1228) to describe the updated functionality?
- Can you an add [example](https://github.com/u01jmg3/ics-parser/blob/master/examples/index.php#L33) similar to `eventsFromRange()`?
- Can you also remove the blank newline at the end of the file - no need for it. Suspect it is your editor adding it in

**Comment by u01jmg3** [comment]
Final thing to fix is when an event does not have an end date set

Suggest using this code in `eventsFromRange()`:

```php
foreach ($events as $anEvent) {
    $eventStart = $anEvent->dtstart_array[2];
    $eventEnd   = (isset($anEvent->dtend_array[2])) ? $anEvent->dtend_array[2] : null;

    if (($eventStart >= $rangeStart && $eventStart < $rangeEnd)        // Event start date contained in the range
        || ($eventEnd !== null)
        && (
               ($eventEnd > $rangeStart && $eventEnd <= $rangeEnd)     // Event end date contained in the range
               || ($eventStart < $rangeStart && $eventEnd > $rangeEnd) // Event starts before and finishes after range
           )
    ) {
        $extendedEvents[] = $anEvent;
    }
}
```

**Comment by grimmlink** [comment]
I agree though i changed the parenthesis for the `$eventEnd !== null` or it would accept any event with an end date :)

```
foreach ($events as $anEvent) {
    $eventStart = $anEvent->dtstart_array[2];
    $eventEnd   = (isset($anEvent->dtend_array[2])) ? $anEvent->dtend_array[2] : null;

    if (($eventStart >= $rangeStart && $eventStart < $rangeEnd)         // Event start date contained in the range
        || ($eventEnd !== null
            && (
                ($eventEnd > $rangeStart && $eventEnd <= $rangeEnd)     // Event end date contained in the range
                || ($eventStart < $rangeStart && $eventEnd > $rangeEnd) // Event starts before and finishes after range
            )
        )
    ) {
        $extendedEvents[] = $anEvent;
    }
}
```

**Comment by u01jmg3** [comment]
> I agree though I changed the parenthesis for the $eventEnd !== null or it would accept any event with an end date

Thanks! :)

**Pull request merged**